### DISCUSSION TITLE
Manual indexing

### DISCUSSION THREAD

**Discussion by smorad** [ymd-h/cpprb#27] [Q&A] [answered]
Is there a way to slice and get a portion of the replay buffer starting at a specific index? Ideally, something like
```python
rb = ReplayBuffer(...)
rb[100:200]
```

There is `rb.get_all_transitions['obs'][100:200]` but this requires materialising the entire replay buffer to get just a sample. Is this efficient?

**Comment by ymd-h** [comment, marked as answer]
@smorad 

Thank you for your feedback.

There is a private but almost stable method `ReplayBuffer._encode_sample(indices)`

https://github.com/ymd-h/cpprb/blob/3b791b59604a7137ed219180169fcd19e1c5e449/cpprb/PyReplayBuffer.pyx#L1321

You can pass list of indices.
